pkg/virtctl/imageupload/util.go

```diff
@@ -0,0 +1,145 @@
+package imageupload
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	instancetypeapi "kubevirt.io/api/instancetype"
+	cdiv1 "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
+)
+
+func (c *command) parseArgs(args []string) error {
+
+	if len(args) != 2 {
+		return fmt.Errorf("expecting two args")
+	}
+
+	switch strings.ToLower(args[0]) {
+	case "dv":
+		c.createPVC = false
+	case "pvc":
+		c.createPVC = true
+	default:
+		return fmt.Errorf("invalid resource type %s", args[0])
+	}
+
+	c.name = args[1]
+
+	return nil
+}
+
+func (c *command) createStorageSpec() (*cdiv1.StorageSpec, error) {
+	quantity, err := resource.ParseQuantity(c.size)
+	if err != nil {
+		return nil, fmt.Errorf("validation failed for size=%s: %s", c.size, err)
+	}
+
+	spec := &cdiv1.StorageSpec{
+		Resources: v1.VolumeResourceRequirements{
+			Requests: v1.ResourceList{
+				v1.ResourceStorage: quantity,
+			},
+		},
+	}
+
+	if c.storageClass != "" {
+		spec.StorageClassName = &c.storageClass
+	}
+
+	if c.accessMode != "" {
+		if c.accessMode == string(v1.ReadOnlyMany) {
+			return nil, fmt.Errorf("cannot upload to a readonly volume, use either ReadWriteOnce or ReadWriteMany if supported")
+		}
+		spec.AccessModes = []v1.PersistentVolumeAccessMode{v1.PersistentVolumeAccessMode(c.accessMode)}
+	}
+
+	switch c.volumeMode {
+	case "block":
+		volMode := v1.PersistentVolumeBlock
```

Comment:
inline this:
spec.VolumeMode = pointer.P(spec.VolumeMode)
The same for filesystem mode